nodebuilder/header/opts.go

```diff
@@ -0,0 +1,17 @@
+package header
+
+import (
+	header "github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
+	p2p "github.com/celestiaorg/celestia-node/libs/header/p2p"
+)
+
+// WithMetrics provides sets `MetricsEnabled` to true on ClientParameters for the header exchange
+func WithMetrics(ex libhead.Exchange[*header.ExtendedHeader]) error {
+	switch p2pex := any(ex).(type) {
+	case *p2p.Exchange[*header.ExtendedHeader]:
+		return p2pex.RegisterMetrics()
+	default:
+		return nil
```

Comment:
if this is called and there's no p2p exchange - shouldn't it return err?

Reply:
No, because of Core.Exchange